rpcserver.go

```diff
@@ -4184,7 +4184,11 @@ func handleGetWorkRequest(s *rpcServer) (interface{}, error) {
 		// Update the time of the block template to the current time
 		// while accounting for the median time of the past several
 		// blocks per the chain consensus rules.
-		UpdateBlockTime(msgBlock, s.server.blockManager)
+		err := UpdateBlockTime(msgBlock, s.server.blockManager)
+		if err != nil {
+			errStr := fmt.Sprintf("Failed to update block time: %v", err)
+			return nil, rpcInternalError(errStr, "")
```

Comment:
The `rpcInternalError` function accepts the context precisely to avoid the need to manually fmt the error.

```Go
return nil, rpcInternalerror(err.Error(), "Failed to update block time")
```